### PR TITLE
Added keep-together paragraph style

### PR DESCRIPTION
**Pull request by verheyenkoen**

### CODE REVIEW

**Comment by connium** [comment]
Hello @verheyenkoen , thank you very much for your contribution!

When reading the code and comparing the output against a document created by LibreOffice I encountered two issues:
1. Please add a possibility to revert the keep-together flag. There might be some scenarios where a document is automatically created and during the generation process a condition is met where you want to change/unset any property (happened to me with the `setPageBreakBefore` method, see issue #31 )
  Maybe by providing a boolean flag as argument to the method (could default to `true`).
1. When I configure LibreOffice to keep together a paragraph, it adds a `<text:soft-page-break/>` at the beginning of the respective paragraph. Do you know whether this is necessary? I wasn't able to find any information neither in the [OASIS specification](http://docs.oasis-open.org/office/v1.2/os/OpenDocument-v1.2-os-part1.html#property-fo_keep-together) nor in the [Relax NG Schema](http://docs.oasis-open.org/office/v1.2/os/OpenDocument-v1.2-os-schema.rng).

File: src/style/ParagraphProperties.ts [hunk 1]
```diff
@@ -16,6 +17,7 @@ const DEFAULT_PAGE_BREAK = false;
 export class ParagraphProperties implements IParagraphProperties {
   private horizontalAlignment: HorizontalAlignment;
   private shouldBreakPageBefore: boolean;
+  private shouldKeepParagraphTogether: boolean;
```
Comment (connium) [inline src/style/ParagraphProperties.ts]: Please rename variable to `shouldKeepTogether`. The JSDoc of the method explains what this setting is meant for and I try to name the properties like they are named in the schema (makes it easier in the `toXml method`), if possible.

File: src/style/ParagraphProperties.ts [hunk 2]
```diff
@@ -127,6 +139,20 @@ export class ParagraphProperties implements IParagraphProperties {
     }
   }
 
+  /**
+   * Sets the keep together attribute if paragraph marked as keep together.
+   *
+   * @param {Element} textPropertiesElement The element which will take the attribute
```
Comment (connium) [inline src/style/ParagraphProperties.ts]: Please fix the name of the attribute (the attribute actually is named `paragraphPropertiesElement`).

I just discovered it's wrong for the other methods in this class as well. If you don't mind, you may adjust the other JSDocs as well.

File: src/style/ParagraphProperties.ts [hunk 3]
```diff
@@ -127,6 +139,20 @@ export class ParagraphProperties implements IParagraphProperties {
...
+   */
+  private setKeepTogetherAttribute(paragraphPropertiesElement: Element): void {
+    if (this.shouldKeepParagraphTogether === true) {
+      paragraphPropertiesElement.setAttribute(
+        OdfAttributeName.FormatKeepTogether,
+        "always"
+      );
```
Comment (connium) [inline src/style/ParagraphProperties.ts]: max line length is 120, so you can put the whole in one line

Comment (verheyenkoen) [reply src/style/ParagraphProperties.ts]: Yeah, that's my local eslint kicking in.

File: src/style/ParagraphProperties.ts (hunk 1 above)
Comment (verheyenkoen) [reply src/style/ParagraphProperties.ts]: Addressed in commit c107b2f0b4449672eb57777dd8417a35827f13c8

File: src/style/ParagraphProperties.ts (hunk 2 above)
Comment (verheyenkoen) [reply src/style/ParagraphProperties.ts]: Addressed in 3326a03e753f8b06119111c104538b78e87dde14

File: src/style/ParagraphProperties.ts (hunk 3 above)
Comment (verheyenkoen) [reply src/style/ParagraphProperties.ts]: Addressed in 8aa67a03bd379c1754f98c13d136d76d23bd89bf

**Comment by verheyenkoen** [comment]
About your main remarks:

1. Addressed in 76f247c22dbb013b6a916957e2de02ac96fc3a00
2. I tested this (without adding `<text:soft-page-break/>`) and seems to work. In LibreOffice the paragraph style setting "Do not split paragraph" is checked and the paragraphs are indeed kept on the same page.

Anything else you think should be addressed for this PR?

**Comment by connium** [comment]
Thank you for improving the code. :sparkles: 

I'll merge your pull request this evening.

**Pull request merged**